torchtune/modules/transformer.py

```diff
@@ -485,7 +497,20 @@ def reset_caches(self):
         for layer in self.layers:
             layer.reset_cache()
 
-    @torch.compiler.disable
+    @property
+    def linear_projection_weight(self) -> torch.Tensor:
+        """Returns the output weight matrix. Useful when a finer control of the output projection is needed,
+        for example when using a custom loss function or when interested in applying it to only some tokens.
+        """
+        # Accessing the weight directly will not trigger FSDP hooks
+        # to gather the full tensor so we have to unshard manually
+        if isinstance(self.output, FSDPModule):
+            self.output.unshard()
+            weight = self.output.weight.clone()
```

Comment:
so the fix was the removal of detach? also any memory implications of the clone here?